src/Mod/Measure/App/MeasureArea.cpp

```diff
@@ -0,0 +1,184 @@
+/***************************************************************************
+ *   Copyright (c) 2023 David Friedli <david[at]friedli-be.ch>             *
+ *                                                                         *
+ *   This file is part of the FreeCAD CAx development system.              *
+ *                                                                         *
+ *   This library is free software; you can redistribute it and/or         *
+ *   modify it under the terms of the GNU Library General Public           *
+ *   License as published by the Free Software Foundation; either          *
+ *   version 2 of the License, or (at your option) any later version.      *
+ *                                                                         *
+ *   This library  is distributed in the hope that it will be useful,      *
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
+ *   GNU Library General Public License for more details.                  *
+ *                                                                         *
+ *   You should have received a copy of the GNU Library General Public     *
+ *   License along with this library; see the file COPYING.LIB. If not,    *
+ *   write to the Free Software Foundation, Inc., 59 Temple Place,         *
+ *   Suite 330, Boston, MA  02111-1307, USA                                *
+ *                                                                         *
+ ***************************************************************************/
+
+
+#include "PreCompiled.h"
+
+#include <App/Application.h>
+#include <App/Document.h>
+
+#include "MeasureArea.h"
+
+
+using namespace Measure;
+
+PROPERTY_SOURCE(Measure::MeasureArea, Measure::MeasureBase)
+
+
+
+MeasureArea::MeasureArea()
+{
+    ADD_PROPERTY_TYPE(Elements,(nullptr), "Measurement", App::Prop_None, "Element to get the area from");
+    Elements.setScope(App::LinkScope::Global);
+    Elements.setAllowExternal(true);
+
+    ADD_PROPERTY_TYPE(Area,(0.0), "Measurement", App::PropertyType(App::Prop_ReadOnly|App::Prop_Output),
+                                            "Area of element");
+
+}
+
+MeasureArea::~MeasureArea() = default;
+
+
+bool MeasureArea::isValidSelection(const App::MeasureSelection& selection){
+
+    if (selection.empty()) {
+        return false;
+    }
+
+    App::Document* doc = App::GetApplication().getActiveDocument();
+
+    for (const std::tuple<std::string, std::string>& element : selection) {
+        const std::string& obName = get<0>(element);
+        App::DocumentObject* ob = doc->getObject(obName.c_str());
+        
+        const std::string& subName = get<1>(element);
+        const char* className = ob->getSubObject(subName.c_str())->getTypeId().getName();
+        std::string mod = ob->getClassTypeId().getModuleName(className);
```

Comment:
That's a weird use of getModuleName(). It's a static method of the class Type sou you can directly use it  as Type::getModuleName()